moveit_ros/moveit_servo/CMakeLists.txt

```diff
@@ -1,146 +1,188 @@
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.5)
 project(moveit_servo)
 
-if(NOT "${CMAKE_CXX_STANDARD}")
-  set(CMAKE_CXX_STANDARD 14)
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
 endif()
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-
-set(LIBRARY_NAME moveit_servo_cpp_api)
 
-if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Release)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
 
-find_package(Boost REQUIRED)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
+set(LIBRARY_NAME moveit_servo)
```

Comment:
I like the idea of using variables like this. Alternatively, we could follow the ROS 1 convention is to use patterns like `${PROJECT_NAME}_lib`, `${PROJECT_NAME}_server`.

Reply:
I don't mind either way. There are other variables set here which don't fit nicely into `${PROJECT_NAME}_lib` or `${PROJECT_NAME}_server`